frontend/shared/src/hocs/initLocale.tsx

```diff
@@ -0,0 +1,17 @@
+import { useRouter } from 'next/router';
+import { useTranslation } from 'next-i18next';
+import React from 'react';
+
+const initLocale = <P,>(
+  WrappedComponent: React.FC<P>
+): typeof WrappedComponent => (props: P) => {
+  const router = useRouter();
```

Comment:
this HOC will preserver the locale when refreshing page. Router will read the lang param from the path and it is updated if needed.